runtime/dart_controller.cc

```diff
@@ -146,11 +153,29 @@ tonic::DartErrorHandleType DartController::RunFromSource(
 void DartController::CreateIsolateFor(const std::string& script_uri,
                                       const uint8_t* isolate_snapshot_data,
                                       const uint8_t* isolate_snapshot_instr,
+                                      const std::vector<uint8_t>& platform_kernel,
                                       std::unique_ptr<UIDartState> state) {
   char* error = nullptr;
-  Dart_Isolate isolate = Dart_CreateIsolate(
-      script_uri.c_str(), "main", isolate_snapshot_data, isolate_snapshot_instr,
-      nullptr, static_cast<tonic::DartState*>(state.get()), &error);
+
+  Dart_Isolate isolate;
+  if (!platform_kernel.empty()) {
+    // Copy kernel bytes so they won't go away after we exit this method.
+    // This is needed because original kernel data has to be available
+    // during code execution.
+    uint8_t* platform_kernel_data = (uint8_t*) malloc(platform_kernel.size());
+    memcpy(platform_kernel_data, platform_kernel.data(), platform_kernel.size());
+
```

Comment:
maybe we should add fields platform_kernel_data and platform_kernel_size to the DartController class and do this copy in the constructor of DartController, that way we would be able to reuse it from there for every call to CreateIsolateFor instead of doing the memcpy everytime. Also we can free the buffer in the destructor of DartController.